File: src/main/java/thecudster/sre/events/PlayerPingEvent.java
package thecudster.sre.events;

import net.minecraftforge.fml.common.eventhandler.Event;

public class PlayerPingEvent extends Event {
    public String sender;
    public String receiver;
    public String reason;
    public PlayerPingEvent(String theSender, String theReceiver) {
        this.receiver = theReceiver;
        this.sender = theSender;
        this.reason = null;
    }
    public PlayerPingEvent(String theSender, String theReceiver, String theReason) {
        this.sender = theSender;
        this.receiver = theReceiver;
        this.reason = theReason;
    }
}

File: src/main/kotlin/thecudster/sre/events/PlayerPingEvent.kt
package thecudster.sre.events

import net.minecraftforge.fml.common.eventhandler.Event

class PlayerPingEvent : Event {
    @JvmField
    var sender: String
    @JvmField
    var receiver: String
    @JvmField
    var reason: String?

    constructor(theSender: String, theReceiver: String) {
        receiver = theReceiver
        sender = theSender
        reason = null
    }

    constructor(theSender: String, theReceiver: String, theReason: String?) {
        sender = theSender
        receiver = theReceiver
        reason = theReason
    }
}